Add va-accordion USWDS v3 variation

powellkerry opened the pull request:
## Chromatic
<!-- This `1719-v3-accordion` is a placeholder for a CI job - it will be updated automatically -->
https://1719-v3-accordion--60f9b557105290003b387cd5.chromatic.com

## Description
Convert existing va-accordion web-component to use USWDS styles and architecture. Maintain backwards compatibility with the previous version.

Sketch Design: https://www.sketch.com/s/a52734dd-00d0-44f1-9c9e-ff4016130e5c/a/OeoJpYv#Inspect
USWDS Design: https://designsystem.digital.gov/components/accordion/

Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1719

## Testing done
- Browser testing in Chrome, Firefox, Safari, Edge
- Axe Dev Tools accessibility 
- VoiceOver testing

## Screenshots
![Screenshot 2023-06-26 at 11 38 37 AM](https://github.com/department-of-veterans-affairs/component-library/assets/1776069/a436a3ef-8b5e-4df6-be60-2bf3100876fe)
![Screenshot 2023-06-26 at 11 39 09 AM](https://github.com/department-of-veterans-affairs/component-library/assets/1776069/85b4bbe6-6b55-427f-a3bc-d4ecdb8cda33)


## Acceptance criteria
- [x] Maintain backwards compatibility
- [x] Pass accessibility tests

## Definition of done
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion-item.tsx:
```diff
@@ -49,6 +50,16 @@ export class VaAccordionItem {
+  /**
+   * The id of the accordion item, required for accessibility. (USWDS only)
+   */
+  @Prop() bordered?: boolean = false;
```
Is the documentation text correct for this prop?

```
  /**
   * The id of the accordion item, required for accessibility. (USWDS only)
   */
  @Prop() bordered?: boolean = false;
```

powellkerry replied:
This was incorrect and I updated it, I also realized that I needed to clean up the unused bordered prop on the va-accordion.tsx file. Great find!

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion.tsx:
```diff
@@ -70,55 +97,71 @@ export class VaAccordion {
+          }
+        })
+      }
+    } else {*/
```
Do we need all of this commented out code?

powellkerry replied:
Removed

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion-item.scss:
```diff
@@ -1,5 +1,56 @@
+
+.usa-accordion__button {
+  position: relative;
+}
```
Are these USWDS style overrides?

```
.usa-accordion__header, .usa-accordion__subheader {
  display: block;
}

.usa-accordion__header ::slotted([slot=icon]), .usa-accordion__subheader ::slotted([slot=subheader-icon]){
  margin-right: 15px;
  width: inherit;
}

.usa-accordion__subheader {
  font-weight: 400;
  font-size: 16px;
  line-height: 26px;
  margin-top: 2px;
}

.usa-accordion__button {
  position: relative;
}
```

powellkerry replied:
The uswds styles do not include a subheader and icons for the header and subheader lines. These are the styles to add them.

jamigibbs replied:
ah! Ok so I guess what is confusing is that it's using the `.usa-accordion` namespace which makes me think it's actually from the USWDS module. But it's really just custom classes we added for styling features that are VA specific.

This might be the first time we've done something like this (add a new feature to the USWDS variation - subheader and icons in this case) so it's an interesting scenario we will probably come across again in other components.

Will mimicking a USWDS class for custom styling make it more or less confusing when we eventually need to reconcile a component for the USWDS team? @Andrew565 Considering you've done some preliminary work for "giving back" these components, what are your thoughts on how we should do this? Would it be easier to identify our custom features by using generic classes, or, create best-guess names that are similar to what's in the modules (but not actually in the modules)? Do we even know if that team wants any of our "enhancements"?

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion.tsx:
```diff
@@ -48,15 +49,41 @@ if (Build.isTesting) {
+   * Optional accordion section heading text. Only used in analytics event. Default is null.
+   */
+  @Prop() sectionHeading?: string = null;
+
```
The diff makes it seems like all of these props are new to this file. I can understand the `uswds` one but is that also right for the others?

powellkerry replied:
This is correct-ish. The only new prop is uswds, but I moved the props to the top of the file instead of having them buried beneath the logic at line 160.

jamigibbs replied:
Oh yes! They really were buried down there. 😂 Good idea to group them together.

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion.tsx:
```diff
@@ -188,30 +222,70 @@ export class VaAccordion {
-              this.expanded
-                ? i18next.t('collapse-all-aria-label')
-                : i18next.t('expand-all-aria-label')
+    const {uswds = false, openSingle} = this;
```
I'm curious why `uswds` has to be set to false here.

powellkerry replied:
This was an artifact from development that I forgot to remove.

jamigibbs commented:
I think something is a little off with the Font Awesome icon override. It looks thinner/smaller:

v1:
![Screenshot 2023-06-27 at 12 42 39 PM](https://github.com/department-of-veterans-affairs/component-library/assets/872479/8eb0c0fe-87a2-4650-bd20-d66ccfe3a66a)

v3:
![Screenshot 2023-06-27 at 12 42 43 PM](https://github.com/department-of-veterans-affairs/component-library/assets/872479/94956138-ecba-4798-8599-7a085f087770)

jamigibbs commented:
A small PR administration request; can we name this PR the way we have for all of the other USWDS v3 variations for our release notes? ie. Add va-accordion USWDS v3 variation

jamigibbs commented:
I was expecting to find `.usa-accordion--bordered .usa-accordion__content` in the formation override stylesheet (because it's full of rem's) but I don't see it there. Do we know where this style is being applied to the border? It looks thinner than what's on the USWDS site:

![Screenshot 2023-06-27 at 12 51 26 PM](https://github.com/department-of-veterans-affairs/component-library/assets/872479/2de3ae9c-a01c-416d-8aba-84d9afecba52)

powellkerry commented:
@jamigibbs Good catch on the content rems. I didn't see that, probably because the v1 styles were still applied. I have fixed it so that the v1 styles are not applied and added the formation overrides.

babsdenney reviewed (approved):
The design looks great!

babsdenney commented:
@powellkerry I was looking at this again and noticed that the USWDS sets a max width for the content inside the accordion. Looking at our previous accordion we didn't have a max-width set. This might be a big visual shift in content for teams using the accordion. 

@humancompanion-usds should we be setting a max width for the content within the accordion?

humancompanion-usds commented:
@babsdenney - Great question. They set it to 64ex to keep it to a reasonable reading length. I think we need a11y to weigh in here. If they would back this change then I'd be all for it despite the repercussions.

jamigibbs reviewed (approved):
Great job @powellkerry!

I think we might just want to do a minor version update and then 🚀

Andrew565 reviewed (approved):
LGTM, great job.
